### DISCUSSION TITLE
identityFile on localhost

### DISCUSSION THREAD

**Discussion by eppak** [deployphp/deployer#2269] [General]
Hello,
i have a github deploy key for a priva repository, i have a deploy.php file like this:

`namespace Deployer;

require 'recipe/common.php';

// Project name
set('application', 'my_project');

// Project repository
set('repository', 'git@github.com:eppak/myrepository.git');

// [Optional] Allocate tty for git clone. Default value is false.
set('git_tty', true);

// Shared files/dirs between deploys
set('shared_files', []);
set('shared_dirs', []);

// Writable dirs by web server
set('writable_dirs', []);

// Hosts
set('allow_anonymous_stats', false);

localhost()
->forwardAgent(true)
->multiplexing(true)
->set('deploy_path', '/home/user/tmp')
->identityFile('/home/user/user/sshkey')
->addSshOption('UserKnownHostsFile', '/dev/null')
->addSshOption('StrictHostKeyChecking', 'no')
->configFile('/home/user/tmp/ssh_config');

// Tasks

desc('Deploy your project');
task('deploy', [
    'deploy:info',
    'deploy:prepare',
    'deploy:lock',
    'deploy:release',
    'deploy:update_code',
    'deploy:shared',
    'deploy:writable',
    'deploy:vendors',
    'deploy:clear_paths',
    'deploy:symlink',
    'deploy:unlock',
    'cleanup',
    'success'
]);`

ssh_config file:

`
Host *
   StrictHostKeyChecking no
   UserKnownHostsFile=/dev/nul
`

Running sudo -u user dep deploy lead me to an error "Exit Code: 128(Invalid exit argument)", this is due to the invalid credentials. But if i modify fingerprint check in /etc/ssh/ssh_config and put sshkey in /home/user/.ssh/id_rsa it works.

What i'm missing? I suppose this is my mistake but i cannot figure why.
Thansk

**Comment by antonmedv** [comment]
What’s the error? Investigate from there.

**Reply by eppak** [reply]
The error is that is not reading the key, it say "git@github.com: Permission denied (publickey)".
I noticed that this is a workaround:

```php
set('env', [
'GIT_SSH_COMMAND' => 'ssh -i /home/user/tmp/id_rsa'
]);
```
This env var is valid for a git 2+ version and it works.

I think that in localhost identityFile and configFile are ignored, is a supposition.
Thansks for your time

**Comment by eppak** [comment]
Any news?

**Reply by antonmedv** [reply]
I bought a nice IKEA cabinet.

**Reply by eppak** [reply]
Good for you. Need just to know if the fact that is ignoring identityFile or configFile in a localhost configuration is a feature of the new shiny cabinet

**Reply by antonmedv** [reply]
The price is good. Love it. Thanks for supporters who donated money to me for Deployer! Localhost does not use ssh at all because it’s localhost.

**Reply by eppak** [reply]
You don't say? But git does, no problem I will use this workaround, but with this kind of kindness you will continue to buy just ikea forniture 😂😜

**Reply by antonmedv** [reply]
Sure. Why are you not supporting Deployer? $1/month will be good.

**Reply by eppak** [reply]
Done.
